server/ot_rpc.go

```diff
@@ -141,7 +142,11 @@ func runTranscode(n *core.LivepeerNode, orchAddr string, httpc *http.Client, not
 // Orchestrator gRPC
 
 func (h *lphttp) RegisterTranscoder(req *net.RegisterRequest, stream net.Transcoder_RegisterTranscoderServer) error {
-	glog.Info("Got a RegisterTranscoder request for ", req.Secret)
+	from := "unknown"
+	if p, ok := peer.FromContext(stream.Context()); ok {
```

Comment:
Very nice! Looking at the `peer` docs in [1] , what does it mean if the stream does not exist in the context? Is the stream disconnected? If so, can we just exit early and not register?

Reply:
If I'm not mistaken, stream and context created [here](https://github.com/grpc/grpc-go/blob/master/internal/transport/http2_server.go#L330), so stream should always have context with peer.

Reply:
Kinda funny how grpc goes against golang's own guidelines not to store contexts in a struct: https://golang.org/pkg/context/